stdlib/ballerina-builtin/src/main/ballerina/ballerina/security/authentication/basic/authn-interceptor.bal

```diff
@@ -0,0 +1,65 @@
+package ballerina.security.authentication.basic;
+
+import ballerina.net.http;
+import ballerina.log;
+import ballerina.security.utils;
+import ballerina.security.crypto;
+
+@Description {value:"Authentication cache name"}
+const string AUTH_CACHE = "basic_auth_cache";
+
+@Description {value:"Basic authenticator instance"}
+BasicAuthenticator authenticator;
+
+@Description {value:"Intercepts a request for authentication"}
+@Param {value:"req: InRequest object"}
+@Return {value:"boolean: true if authentication is a success, else false"}
+public function handle (http:InRequest req) (boolean) {
+    // extract the header value
+    string basicAuthHeaderValue;
+    error err;
+    basicAuthHeaderValue, err = utils:extractBasicAuthHeaderValue(req);
+    if (err != null) {
+        log:printErrorCause("Error in extracting basic authentication header", err);
+        return false;
+    }
+
+    if (authenticator == null) {
+        authenticator = createAuthenticator({}, utils:createCache(AUTH_CACHE));
+    }
+    AuthenticationInfo authInfo;
+    // check in the cache - cache key is the sha256 hash of the basic auth header value
+    string basicAuthCacheKey = crypto:getHash(basicAuthHeaderValue, crypto:Algorithm.SHA256);
+    any cachedAuthResult = authenticator.getCachedAuthResult(basicAuthCacheKey);
+    if (cachedAuthResult != null) {
+        // cache hit
+        var authInfo, typeCastErr = (AuthenticationInfo)cachedAuthResult;
+        if (typeCastErr == null) {
+            // no type cast error, return cached result.
+            log:printDebug("Auth cache hit for user: " + authInfo.username);
+            return authInfo.isAuthenticated;
+        }
+        // if a casting error occurs, clear the cache entry
+        authenticator.clearCachedAuthResult(basicAuthCacheKey);
+    }
+
+    string username;
+    string password;
+    username, password, err = utils:extractBasicAuthCredentials(basicAuthHeaderValue);
```

Comment:
Cant we use `var username, password, err = utils:extractBasicAuthCredentials(basicAuthHeaderValue);` here?